cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-java/src/main/java/com/sap/cloud/cf/monitoring/java/converter/GaugeConverter.java

```diff
@@ -0,0 +1,25 @@
+package com.sap.cloud.cf.monitoring.java.converter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+
+import com.codahale.metrics.Gauge;
+import com.sap.cloud.cf.monitoring.client.model.Metric;
+
+public final class GaugeConverter extends MetricConverter<Gauge> {
+
+    @Override
+    protected List<Metric> convertMetricEntry(Entry<String, Gauge> metricEntry, long timestamp) {
+        ArrayList<Metric> result = new ArrayList<>();
+        Object gaugeValue = metricEntry.getValue().getValue();

```

Comment:
Just out of curiosity: Is `metricEntry.getValue()` guaranteed to always return a non-nil value?

Reply:
getMetrics method in com.codahale.metrics.MetricRegistry checks the entry's value, so if there is an java.util.Map.Entry<String, Gauge> its value should not be null.